source/request-as-event-emitter.ts

```diff
@@ -233,24 +243,25 @@ export default (options, input?: TransformStream) => {
 
 			cacheRequest.once('request', handleRequest);
 		} else {
-			// Catches errors thrown by calling fn.request(...)
+			// Catches errors thrown by calling requestFn(...)
```

Comment:
It's `requestFn.request` if you do it this way. IMO it should be `fn.request` as before, no need to change.

Reply:
While the old `fn` was an object with a `request` property, the new `requestFn` is the request function itself. Typing the `fn` object would look like

```ts
interface ObjWithRequestFunction {
  request: RequestFunction
}
```

and would still require creating a duck-typed object with a `request` property:

https://github.com/sindresorhus/got/pull/760/files#diff-267e4380ccf0cfebc8b885a948acf226L62

All of that seems unnecessarily indirect.

The diff is a bit confusing, mostly because `fn` was a confusing name, though the new code seems clearer and more direct.

Reply:
Ah. After that commit it makes sense now